pages/previewPage.html

```diff
@@ -0,0 +1,174 @@
+<!DOCTYPE html>
+<html lang="en" title="d2 Diagram">
+<title>Diagram</title>
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
+<meta charset="utf-8">
+
+<head>
+    <style>
+        #toolbar {
+            z-index: 100;
+            position: fixed;
+            left: 0px;
+            top: 0px;
+            background-color: #909090;
+            padding-left: 15px;
+            padding-right: 200px;
+            padding-bottom: 5px;
+            width: 100%;
+            color: #0D32B2;
+        }
+
+        #zoomSlider {
+            appearance: none;
+            border-radius: 3px;
+            background: #0D32B2;
+            height: 5px;
+        }
+
+        #zoomSlider::-webkit-slider-thumb {
+            appearance: none;
+            box-shadow: 1px 1px 1px #000000, 0px 0px 1px #0d0d0d;
+            border: 1px solid darkblue;
+            height: 20px;
+            width: 8px;
+            margin-bottom: 5px;
+            border-radius: 3px;
+            background: lightblue;
+            cursor: pointer;
+        }
+
+        #percent {
+            display: inline-block;
+            font-family: monospace;
+            font-size: 18px;
+            font-weight: 600;
+            width: 55px;
+        }
+
+        button {
+            margin-top: 3px;
+            color: #0D32B2;
+            border-color: #0D32B2;
+            background-color: #F7F8FE;
+            border-radius: 6px;
+            border: 3 3 3 3;
+            box-shadow: 1px 1px 1px black, 0px 0px 1px 0d0d0d;
+            padding: 5px;
+            min-width: 50px;
+            font-size: 14px;
+            font-weight: 600;
+        }
+
+        svg {
+            position: absolute;
+            left: 0px;
+            top: 0px;
+            transform-origin: 0px 0px;
+        }
+
+        body {
+            background-color: var(--vscode-editor-background);
+            color: var(--vscode-editor-foreground);
+            fill: var(--vscode-editor-foreground);
+            width: 100%;
+            height: 100%;
+        }
+    </style>
+    <script>
+        var zoomLevel = 100;
+        var percent;
+        var zoomSlider;
+        var fitMode = true;
+
+        function getSvgElement() {
+            var svgs = document.getElementsByTagName('svg');
+            return svgs[0];
+        }
+        function init() {
+            percent = document.getElementById('percent');
+            zoomSlider = document.getElementById('zoomSlider');
+            zoomFit();
+        }
+        function zoomPlus() {
+            zoomLevel = Math.min(150, zoomLevel + 5);
+            setZoomLevel(zoomLevel);
+        }
+        function zoomMinus() {
+            zoomLevel = Math.max(10, zoomLevel - 5);
+            setZoomLevel(zoomLevel);
+        }
+        function zoomFit() {
+            setZoomLevel(100);
+            var svg = getSvgElement();
+            svg.style.height = "100%";
```

Comment:
I think in the future we will want to be able to also update the zoom level with `fit` so you can fit and then zoom in or out relative to that. This can be a follow up issue though.

Reply:
Yes, I want that too.

Reply:
Fixed in PR #16.  How dare you make me do math...